Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -2840,7 +2840,12 @@ void RenderLayer::paintSVGResourceLayer(GraphicsContext& context, const AffineTr
 
     auto localPaintDirtyRect = LayoutRect::infiniteRect();
 
-    auto* rootPaintingLayer = enclosingSVGRootLayer();
+    auto* rootPaintingLayer = [this] () {
+        auto* curr = parent();
+        while (curr && !(curr->renderer().isAnonymous() && is<RenderSVGViewportContainer>(curr->renderer())))
```

Comment:
Great, that this fixes the problem, without having to resort to context translations.

For completeness, we should duplicate the testcase that you fixed, and alter it to have a viewBox on the <svg> that effectively induces e.g. a scale(2) translation, to make sure paintSVGResourceLayer() is correct in these scenarios

Reply:
I added the test, however I chose a smaller scale so all 3 columns still fit (I think r-w-t limits to 450x450 or so). Still need to see if platforms like gtk produce different txt/png results.